lib/distribute_reads.rb

```diff
@@ -84,7 +84,15 @@ def self.lag(connection: nil)
           status ? status["Replica_lag_in_msec"].to_f / 1000.0 : 0.0
         else
           status = connection.exec_query("SHOW SLAVE STATUS").to_hash.first
-          status ? status["Seconds_Behind_Master"].to_f : 0.0
+          if status
+            if status["Seconds_Behind_Master"].to_s == 'NULL'
```

Comment:
Seems like this would be `nil?` instead of the string `NULL`. Also, I think we should have the `lag` function return `nil` in this case instead of -1 (if -1 is just something AWS does).

Reply:
While debugging, it may be good check for `nil?`, empty string, and the string `NULL` and report what it actually returns.

Reply:
Ok cool yes will look into this.

Reply:
Yes we confirmed NULL does get turned to nil on to_hash call.